plugins/texteditor.koplugin/main.lua

```diff
@@ -0,0 +1,164 @@
+        },
+        {
+            text = _("Close"),
+            callback = function()
+                UIManager:show(ConfirmBox:new{
+                    text = _("Are you sure that you want to close the editor? All unsaved changes will be lost."),
+                    ok_text = _("Close"),
+                    ok_callback = function()
+                        self.context = ""
+                        self.file_path = ""
+                        UIManager:close(self.input)
+                    end
+                })
+
+            end,
+        },
+        {
+            text = _("Pg up"),
+            callback = function()
+                self.input._input_widget.text_widget:scrollUp()
+                UIManager:setDirty(self.input, function()
+                    return "ui", self.input.dimen
+                end)
+            end,
+        },
+        {
+            text = _("Pg dn"),
+            callback = function()
+                self.input._input_widget.text_widget:scrollDown()
+                UIManager:setDirty(self.input, function()
+                    return "ui", self.input.dimen
+                end)
+            end,
+        }
+
+    }},
+    }
+    --to be able to insert new line, we need to disable callback (InputDialog always sets it)
+    self.input._input_widget.enter_callback = nil
+    UIManager:show(self.input)
+    self.input:onShowKeyboard()
+end
+
+function TextEditor:chooseFile()
+    self.input:onClose()
+    local path_chooser = PathChooser:new{
+        title = _("Open file. Long press to confirm"),
+        height = Screen:getHeight(),
+        path = util.realpath(DataStorage:getDataDir()),
+        show_hidden = G_reader_settings:readSetting("show_hidden"),
+        file_filter = function() return true end,
+        onConfirm = function(file_path)
+            logger.dbg("TextEditor: selected file_file_path " .. file_path )
+            self.file_path = file_path
+            self:readFile(self.file_path)
+            self.input:onShowKeyboard()
+        end
+    }
+    UIManager:show(path_chooser)
+end
+
+function TextEditor:readFile(file_path)
+    logger.dbg("TextEditor: reading file: " .. file_path)
+    local file = io.open(file_path, "rb")
+    if file then
+        self.file_path = file_path
+        self.context = file:read("*all")
+        self.input:setInputText(self.context)
+        file:close()
+    else
+        UIManager:show(InfoMessage:new{
+            text = _("Failed to read file: \n" .. file_path)
+        })
+    end
+    UIManager:forceRePaint()
+end
+
+function TextEditor:saveFile(file_path)
+    logger.dbg("TextEditor: saving file: " .. file_path)
+    local file = io.open(file_path, "w")
+    if file then
+        UIManager:show(ConfirmBox:new{
+            text = _("Are you sure that you want to save changes to file: \n" .. file_path),
+            ok_text = _("Save"),
+            ok_callback = function()
+                file:write(self.input:getInputText())
```

Comment:
You might want to do a quick sanity check on the text too before starting to write things.